homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -573,3 +579,29 @@ async def set_room_setpoint(self, temp) -> float:
         return await self.gateway.set_target_temp(
             temp, self.options.get(CONF_TEMPORARY_OVRD_MODE, True)
         )
+
+    async def set_control_setpoint(self, circuit: int, setpoint: float) -> float:
+        """Set the control setpoint on a heating circuit. Return the new setpoint."""
+        if circuit == 2:
+            ret_value = await self.gateway.set_control_setpoint_2(setpoint)
+            key = ENTITY_KEY_CENTRAL_HEATING_2_OVERRIDE
+        else:
+            ret_value = await self.gateway.set_control_setpoint(setpoint)
+            key = ENTITY_KEY_CENTRAL_HEATING_1_OVERRIDE
+        ent_reg = er.async_get(self.hass)
+        ent_id = ent_reg.async_get_entity_id(
+            SWITCH_DOMAIN,
+            DOMAIN,
+            f"{self.hub_id}-{OpenThermDeviceIdentifier.GATEWAY}-{key}",
+        )
+
+        if ent_id is None or (entity := ent_reg.async_get(ent_id)) is None:
+            return ret_value
+
+        if not entity.disabled and ret_value is not None:
+            switch_service = SERVICE_TURN_OFF if ret_value == 0 else SERVICE_TURN_ON
+            await self.hass.services.async_call(
+                SWITCH_DOMAIN, switch_service, {ATTR_ENTITY_ID: ent_id}
+            )
```

Comment:
So why do we do this exactly?

Reply:
It is something that the gateway does internally, as per the [docs](http://otgw.tclcode.com/firmware.html#configuration) of the `CS` and `C2` commands.
Unfortunately the gateway does not have a way to retrieve the current status of the override, so this is a way to sync the switch status with the current override status.
Calling the service on the switch here also sends another override on/off command to the gateway, but since the gateway has already updated the setting with the `CS` or `C2` command this won't change the actual behavior.